src/declaration.d

```diff
@@ -128,12 +128,16 @@ enum STCautoref             = (1L << 45);   // Mark for the already deduced 'aut
 enum STCinference           = (1L << 46);   // do attribute inference
 enum STCexptemp             = (1L << 47);   // temporary variable that has lifetime restricted to an expression
 enum STCmaybescope          = (1L << 48);   // parameter might be 'scope'
+enum STCfuture              = (1L << 49);   // introducing new base class function
```

Comment:
I would say this is/should be more general than base class function. Any symbol could be declared `@future`. If this is just to start with a simple implementation that covers a big percentage of the uses for `@future`, then OK, I'm cool with that, but maybe it should be mentioned.